service/raft/group0_state_machine.cc

```diff
@@ -53,11 +54,42 @@ namespace service {
 
 static logging::logger slogger("group0_raft_sm");
 
+class group0_state_id_handler : public enable_shared_from_this<group0_state_id_handler> {
+
+    std::reference_wrapper<gms::gossiper> _gossiper;
```

Comment:
Why is reference_wrapper needed over a regular reference?

Reply:
As discussed, it was because of the clang-tidy giving a warning about the reference members: https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/avoid-const-or-ref-data-members.html

> Having such members is rarely useful, and makes the class only copy-constructible but not copy-assignable.

When using the `boost::non_copyable` the warning disappears ... but I didn't see that one being used widely in the source code.

Reply:
The vanilla way would be to manually delete the assignment operators:
`group0_state_id_handler& operator=(const group0_state_id_handler&) = delete;`

this is probably how `boost::non_copyable` silences the warning

but using `boost::non_copyable` is nice too